helper.py

```diff
@@ -490,3 +554,178 @@ def get_mode_velocity(velocity_array_framewise):
             vel_array_all.append(velocity)
     vel_mode = statistics.mode(vel_array_all)
     return vel_mode
+
+# Neural Network Implementation
+
+
+# Loss calculation including MSE and PINN loss
+class SolveEquationLoss(tf.keras.losses.Loss):
+    def __init__(self, X_train, L_R_array, mse_weight=0.5):
+        super(SolveEquationLoss, self).__init__()
+        self.i = 0
+        self.X_train = np.squeeze(X_train, axis = -1)#.tolist()
+        self.L_R_array = L_R_array
+        self.mse_weight = mse_weight
+
+    def call(self, y_true, y_pred):
+        # PINN loss calculation
+        pointCloudProcessCFG = PointCloudProcessCFG()
+        peaks_min_intensity_threshold = find_peaks_in_range_data(self.X_train[self.i], pointCloudProcessCFG, intensity_threshold=100)
+        calculated_value = get_velocity(self.X_train[self.i], peaks_min_intensity_threshold, self.L_R_array[self.i])
+        self.i = self.i+1
+        pinn_loss = tf.reduce_mean(tf.square(calculated_value - y_pred))
+
+        # MSE loss calculation
+        mse_loss = tf.reduce_mean(tf.square(y_true - y_pred))
+
+        # Combined PINN loss and MSE loss
+        combined_loss = (1 - self.mse_weight) * pinn_loss + self.mse_weight * mse_loss
+
+        return combined_loss
```

Comment:
@codingsneha don't add anything related to PINN in this PR. 

Will merge PINN n the next PR.

Reply:
Got it, updated.